diag_manager/fms_diag_yaml.F90

```diff
@@ -1,62 +1,328 @@
+type (diag_yaml_object) :: diag_yaml  !< Obj containing the contents of the diag_table.yaml
 
+!> @addtogroup fms_diag_yaml_mod
+!> @{
 contains
 
-!> \brief Returns the basedate as an integer array
+!> @brief Returns the basedate as an integer array
 pure function get_basedate (diag_yaml) result (diag_basedate)
-class (diag_yaml_object), intent(in) :: diag_yaml               !< The diag_yaml
-integer, dimension (basedate_size) :: diag_basedate !< Basedate array result to return
-diag_basedate = diag_yaml%diag_basedate
+  class (diag_yaml_object), intent(in) :: diag_yaml               !< The diag_yaml
+  integer, dimension (basedate_size) :: diag_basedate !< Basedate array result to return
+
+  diag_basedate = diag_yaml%diag_basedate
 end function get_basedate
-!> \brief Returns the title of the diag table as an allocated string
+
+!> @brief Returns the title of the diag table as an allocated string
 pure function get_title (diag_yaml) result (diag_title)
-class (diag_yaml_object), intent(in) :: diag_yaml      !< The diag_yaml
-character(len=:),allocatable :: diag_title !< Basedate array result to return
- diag_title = diag_yaml%diag_title
+  class (diag_yaml_object), intent(in) :: diag_yaml      !< The diag_yaml
+  character(len=:),allocatable :: diag_title !< Basedate array result to return
+
+  diag_title = diag_yaml%diag_title
 end function get_title
 
-!> \brief Compares two field type variables
-pure logical function is_field_type_null (in1)
-type(diag_fields_type), intent(in) :: in1
-is_field_type_null = .true.
-end function is_field_type_null
-
-!!TODO
-!> \brief looks for a diag_field based on it's name.
-!! Returns null if field is not found.
-!type(diag_fields_type)function get_diag_table_field (field_name) result (field)
-! character(len=*), intent(IN) :: field_name
-! integer :: i
-! do i = 1,size(diag_fields)
-!     if (trim(field_name) == trim(fms_c2f_string(diag_fields(i)%fname))) then
-!          field = diag_fields(i)
-!write (6,*) field_name//" Found"
-!
-!          return
-!     endif
-! enddo
-! field = null_field_type
-!
-!end function get_diag_table_field
+!> @brief Uses the yaml_parser_mod to read in the diag_table and fill in the
+!! diag_yaml object
+subroutine diag_yaml_object_init
+  integer              :: diag_yaml_id !< Id for the diag_table yaml
+  integer              :: nfiles       !< Number of files in the diag_table yaml
+  integer, allocatable :: file_ids(:)  !< Ids of the files in the diag_table yaml
+  integer              :: i, j         !< For do loops
+  integer              :: total_nvars  !< The total number of variables in the diag_table yaml
+  integer              :: var_count    !< The current number of variables added to the diag_yaml obj
+  integer              :: nvars        !< The number of variables in the current file
+  integer, allocatable :: var_ids(:)   !< Ids of the variables in diag_table yaml
+
+  diag_yaml_id = open_and_parse_file("diag_table.yaml")
```

Comment:
can we rename this to diag_yaml or diag.yaml?  I don't want it to get confsed with the diag_table and its format

Reply:
For posterity, it was decided to use the `diag_table.yaml` name by the FMS group.